### PR TITLE
[SPARK-25489][ML][TEST] Refactor UDTSerializationBenchmark

### PR DESCRIPTION
**Pull request by seancxmao**
## What changes were proposed in this pull request?
Refactor `UDTSerializationBenchmark` to use main method and print the output as a separate file.

Run blow command to generate benchmark results:

```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "mllib/test:runMain org.apache.spark.mllib.linalg.UDTSerializationBenchmark"
```

## How was this patch tested?
Manual tests.

### CODE REVIEW

File: mllib/src/test/scala/org/apache/spark/mllib/linalg/UDTSerializationBenchmark.scala [hunk 1]
```diff
@@ -18,52 +18,52 @@
 package org.apache.spark.mllib.linalg
 
 import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
-import org.apache.spark.util.Benchmark
+import org.apache.spark.util.{Benchmark, BenchmarkBase => FileBenchmarkBase}
 
 /**
  * Serialization benchmark for VectorUDT.
+ * To run this benchmark:
+ * 1. without sbt: bin/spark-submit --class <this class> <spark mllib test jar>
```
Comment (wangyum) [inline mllib/src/test/scala/org/apache/spark/mllib/linalg/UDTSerializationBenchmark.scala]: I think `<` should replaced to `[`:
```scala
[error] /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/java/org/apache/spark/sql/DatasetBenchmark.java:5: error: unknown tag: this
[error]  * 1. without sbt: bin/spark-submit --class <this class> <spark sql test jar>
[error]                                             ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/java/org/apache/spark/sql/DatasetBenchmark.java:5: error: unknown tag: spark
[error]  * 1. without sbt: bin/spark-submit --class <this class> <spark sql test jar>
[error]                                                          ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/java/org/apache/spark/sql/DatasetBenchmark.java:6: error: unknown tag: this
[error]  * 2. build/sbt "sql/test:runMain <this class>"
[error]                                   ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/java/org/apache/spark/sql/DatasetBenchmark.java:7: error: unknown tag: this
[error]  * 3. generate result: SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain <this class>"
[error]   
```

Comment (dongjoon-hyun) [reply mllib/src/test/scala/org/apache/spark/mllib/linalg/UDTSerializationBenchmark.scala]: +1 for fix the docs to pass Jenkins.
Also, could you rebase this PR to resolve conflicts, @seancxmao ?

**Review by wangyum — APPROVED** [review]

File: mllib/src/test/scala/org/apache/spark/mllib/linalg/UDTSerializationBenchmark.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply mllib/src/test/scala/org/apache/spark/mllib/linalg/UDTSerializationBenchmark.scala]: Thank you, @seancxmao . Happy festival! :)

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

Thank you, @seancxmao .

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Pull request closed without merging**